src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,88 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final double LOAD_FACTOR = 0.75;
+    private static final int INITIAL_CAPACITY = 16;
+    private int size;
+    private Node<K, V>[] nodes;
+
+    public MyHashMap() {
+        nodes = new Node[INITIAL_CAPACITY];
+    }
+
+    private class Node<K, V> {
+        private K key;
+        private V value;
+        private Node<K, V> next;
+    }
 
     @Override
     public void put(K key, V value) {
-
+        Node node = new Node();
+        node.key = key;
+        node.value = value;
+        node.next = null;
+        int bucketIndex = findingBucketIndex(key);
+        if (nodes[bucketIndex] == null) {
+            nodes[bucketIndex] = node;
+        } else {
+            Node tempNode = nodes[bucketIndex];
+            Node tempNodePrev = null;
+            while (tempNode != null) {
+                tempNodePrev = tempNode;
+                if (key == null && tempNode.key == null
+                        || key != null && key.equals(tempNode.key)) {
+                    tempNode.value = value;
+                    return;
+                }
+                tempNode = tempNode.next;
+            }
+            tempNodePrev.next = node;
+        }
+        size++;
+        resize();
     }
 
     @Override
     public V getValue(K key) {
+        for (int i = 0; i < nodes.length; i++) {
+            Node temp = nodes[i];
+            while (key == null || temp != null) {
```

Comment:
```suggestion
            while (temp != null) {
```

why do we ned this check(key == null)?